diff --git a/src/main/java/nl/tudelft/ti2206/game/Game.java b/src/main/java/nl/tudelft/ti2206/game/Game.java
index b9d7cea..9118778 100644
--- a/src/main/java/nl/tudelft/ti2206/game/Game.java
+++ b/src/main/java/nl/tudelft/ti2206/game/Game.java
@@ -1,6 +1,7 @@
 package nl.tudelft.ti2206.game;
 
 import nl.tudelft.ti2206.gameobjects.Grid;
+import nl.tudelft.ti2206.gameobjects.OverlayDisplay;
 import nl.tudelft.ti2206.gameobjects.ScoreDisplay;
 import nl.tudelft.ti2206.gameobjects.Tile;
 import nl.tudelft.ti2206.handlers.AssetHandler;
@@ -14,8 +15,14 @@ import com.badlogic.gdx.scenes.scene2d.Group;
 import com.badlogic.gdx.scenes.scene2d.Stage;
 import com.badlogic.gdx.utils.viewport.ScreenViewport;
 
-public class TwentyFourtyGame implements ApplicationListener {
+public class Game implements ApplicationListener {
 
+	public static enum GameState {
+		RUNNING, LOST, WON, CONTINUEING
+	}
+	
+	private static GameState curState;
+	
 	/** The width of the game */
 	public static final int GAME_WIDTH = 600;
 	/** The height of the game */
@@ -24,6 +31,7 @@ public class TwentyFourtyGame implements ApplicationListener {
 	private Stage stage;
 	private Group group;
 	private ScoreDisplay scores;
+	private OverlayDisplay overlays;
 
 	@Override
 	public void create() {
@@ -48,7 +56,10 @@ public class TwentyFourtyGame implements ApplicationListener {
 		}
 
 		scores = new ScoreDisplay(grid);
+		overlays = new OverlayDisplay();
+		// add score displays and overlay displays
 		group.addActor(scores);
+		group.addActor(overlays);
 		// add grid and tiles
 		stage.addActor(group);
 		// add buttons
@@ -85,4 +96,12 @@ public class TwentyFourtyGame implements ApplicationListener {
 	@Override
 	public void resume() {
 	}
+	
+	public static GameState getState() {
+		return curState;
+	}
+	
+	public static void setState(GameState state) {
+		curState = state;
+	}
 }
diff --git a/src/main/java/nl/tudelft/ti2206/game/Launcher.java b/src/main/java/nl/tudelft/ti2206/game/Launcher.java
index 84d8c1b..8a83454 100644
--- a/src/main/java/nl/tudelft/ti2206/game/Launcher.java
+++ b/src/main/java/nl/tudelft/ti2206/game/Launcher.java
@@ -62,6 +62,6 @@ public class Launcher {
 	 * Launches a new LwjglApplication, containing the game.
 	 */
 	public void launch() {
-		new LwjglApplication(new TwentyFourtyGame(), config);
+		new LwjglApplication(new Game(), config);
 	}
 }
diff --git a/src/main/java/nl/tudelft/ti2206/gameobjects/Grid.java b/src/main/java/nl/tudelft/ti2206/gameobjects/Grid.java
index 40eb59d..c62a3cf 100644
--- a/src/main/java/nl/tudelft/ti2206/gameobjects/Grid.java
+++ b/src/main/java/nl/tudelft/ti2206/gameobjects/Grid.java
@@ -4,6 +4,8 @@ import java.util.ArrayList;
 import java.util.List;
 import java.util.Random;
 
+import nl.tudelft.ti2206.game.Game;
+import nl.tudelft.ti2206.game.Game.GameState;
 import nl.tudelft.ti2206.handlers.AssetHandler;
 
 import com.badlogic.gdx.Gdx;
@@ -12,8 +14,6 @@ import com.badlogic.gdx.graphics.g2d.Batch;
 import com.badlogic.gdx.graphics.g2d.TextureRegion;
 import com.badlogic.gdx.scenes.scene2d.Actor;
 
-//import old.handlers.TileHandler;
-
 /**
  * This class represents the 4x4 grid you see when playing 2048.
  * 
@@ -89,6 +89,9 @@ public class Grid extends Actor {
 		} else {
 			initEmptyGrid();
 		}
+
+		// after loading the grid, start the game.
+		Game.setState(GameState.RUNNING);
 	}
 
 	/**
@@ -164,6 +167,10 @@ public class Grid extends Actor {
 		// t.update();
 		// }
 		updateHighestTile();
+		if (Game.getState() == GameState.RUNNING && highestTile == 2048)
+			Game.setState(GameState.WON);
+		if (Game.getState() == GameState.RUNNING && getPossibleMoves() == 0)
+			Game.setState(GameState.LOST);
 	}
 
 	/**
diff --git a/src/main/java/nl/tudelft/ti2206/handlers/ButtonHandler.java b/src/main/java/nl/tudelft/ti2206/handlers/ButtonHandler.java
index 6ee3457..38f6772 100644
--- a/src/main/java/nl/tudelft/ti2206/handlers/ButtonHandler.java
+++ b/src/main/java/nl/tudelft/ti2206/handlers/ButtonHandler.java
@@ -1,6 +1,6 @@
 package nl.tudelft.ti2206.handlers;
 
-import nl.tudelft.ti2206.game.TwentyFourtyGame;
+import nl.tudelft.ti2206.game.Game;
 
 import com.badlogic.gdx.scenes.scene2d.Actor;
 import com.badlogic.gdx.scenes.scene2d.ui.Button;
@@ -9,7 +9,7 @@ import com.badlogic.gdx.scenes.scene2d.utils.ChangeListener;
 public class ButtonHandler {
 
 	/** The width of the game */
-	private static final int GAME_WIDTH = TwentyFourtyGame.GAME_WIDTH;
+	private static final int GAME_WIDTH = Game.GAME_WIDTH;
 	/** The gap between the edge of the game and the button. */
 	private static final int GAP = 15;
 	/** The button width */
